### ISSUE TITLE
Running oxlint in pre-commit hook with lint-staged

### ISSUE THREAD

**Issue by JoseBuendiaDigio** [oxc-project/oxc#1124]
When I use `npx oxlint --fix` in list-staged and oxlint find errors it doesn't throws it. And seems that all is going well.

Maybe is due to in oxlint is a warning and I want that be an error, can I change this?

**Comment by Boshen** [comment]
It seems like there is a problem with lint-staged passing in absolute paths.

A workaround right now is to call "oxlint ." from a function callback, e.g. https://github.com/web-infra-dev/rspack/blob/3c0c6ca9d9cc5f99919520288006e33f14867e57/lint-staged.config.mjs#L7C2-L7C39

I know this is troublesome, but I'll try and fix this when I get the time.

**Comment by Boshen** [comment]
Oh I remember now! There is a rabbit hole with eslint + lint-staged + .eslintignore

* https://github.com/lint-staged/lint-staged#example-use-relative-paths-for-commands
* https://github.com/lint-staged/lint-staged#how-can-i-ignore-files-from-eslintignore
* `https://github.com/eslint/eslint/issues/12206`

tldr when you modify a `.eslintignor`ed file, we will still lint it because the the directory walker's default assumption is that you want to lint this file because the path is passed in explicitly.

**Comment by clarkf** [comment]
So it seems like [lint-staged](https://github.com/lint-staged/lint-staged/) doesn't have much to do with this. It just passes the changed files explicitly to the linting command (e.g. `eslint`, `oxlint`). eslint/eslint#12206's author [was confused about](https://github.com/eslint/rfcs/pull/38#discussion_r334270836) the default behavior, which I totally get.

eslint v8.56.0 (latest version at time of writing) will emit a warning when an `.eslintignore`d file is explicitly passed as a CLI argument, but *will not* lint the file:

```sh
$ ./node_modules/.bin/eslint test.js

./oxlint-test/test.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

✖ 1 problem (0 errors, 1 warning)

$ echo $?
0
```

oxlint (main, as of 2ac26303e0a4f5382dfe51039e8924e0ff32b58a) lets provided paths take precedence over ignore directives and *will lint* the files:

```sh
$ ~/src/oxc/target/debug/oxlint test.js

  ⚠ eslint(no-self-assign): this expression is assigned to itself
   ╭─[test.js:4:1]
 4 │
 5 │ foo = foo;
   ·       ───
   ╰────

Finished in 4ms on 1 file with 78 rules using 8 threads.
Found 1 warning and 0 errors.

$ echo $?
0
```

Had that been an error rather than a warning, the exit code would probably have been `1`, which is different behavior from eslint.

I think there's value in compatibility with eslint. The current behavior of oxlint requires users recreate the ignore logic in their hooks/CI, which seems a bit silly to me.

Would the addition of a `--no-ignore` flag be a good solution here? Happy to work on it if that's the correct direction.

**Comment by Boshen** [comment]
> The current behavior of oxlint requires users recreate the ignore logic in their hooks/CI, which seems a bit silly to me.

I agree.

> Would the addition of a --no-ignore flag be a good solution here? 

Feel free to add any eslint features.

**Comment by Boshen** [comment]
It seems like there are multiple solutions to this. Feel free to create another issue if there are other usages problems.
